### DISCUSSION TITLE
Relationship to react-query

### DISCUSSION THREAD

**Discussion by genu** [DamianOsipiuk/vue-query#228] [Q&A]
Hi @DamianOsipiuk, whats the relationship between `vue-query` and the `@tanstack` project.

I know they plan to create a vue adapter and such down the road, and it looks like you're involved with that project as well -- would this effort be moved there as an adapter, or whats the high level roadmap?

**Comment by DamianOsipiuk** [comment]
Hi,
Yes, the idea is to move this project as an adapter under https://github.com/TanStack/query. 
First I want to bring back proper ESM support there, which was lost during transition to monorepo.

**Reply by genu** [reply]
ok got it. thx

I'm using `vue-query` in a nuxt3 setup. So far no issue, so I think this is pretty stable already.

Looking forward to testing (or contributing to) anything that comes out of the `@tanstack` vue adapter effort.

**Comment by aaronksaunders** [comment]
so it would be safe to use this in a production project and hopefully an "easy" migration when it gets moved under TanStack?

**Reply by DamianOsipiuk** [reply]
Correct.
If you start with v2, the only thing that you would need to do is to change the import statement.
If you start with v1, make sure that you write all of your `queryKey`s as arrays for easier migration.

**Comment by ebisbe** [comment]
I was searching this question as I read something on an issue. I've never heard of the TanStack project so I'm a bit confused as this project was like a 'fork' for Vue from the react-query. Does that means that the react-query is also moved to TanStack?

**Reply by DamianOsipiuk** [reply]
Hi, yes. react-query got rebranded to [@tanstack/react-query](https://github.com/TanStack/query).
Also the framework-agnostic core got released as a separate package ([@tanstack/query-core](https://www.npmjs.com/package/@tanstack/query-core)) and repository restructured to a monorepo.
This enables adapters for different frameworks to be moved to the same repository.

**Reply by ebisbe** [reply]
Thanks. That's great news for everyone!
